xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
Sorry, I don't understand. If the ID gets changed from zero to non zero or the other way round, the result of the expression changes. So the the value of m_ThreadId is important and should be synchronized, I believe.

And I noticed, that m_ThreadId isn't synchronized in the ThreadId() method too. I think this should be changed also. If a mutex is to heavy, maybe m_ThreadId could be changed to std::atomic<int>?